src/ok.rs

```diff
@@ -581,3 +587,51 @@ fn memlock_rlimit() -> TestResult {
         mesg: Some(mesg),
     }
 }
+
+const SEV_CPU_IDS: [&str; 3] = ["7551", "7451", "7401"];
+const ES_CPU_IDS: [&str; 2] = ["7402", "7742"];
+const SNP_CPU_IDS: [&str; 3] = ["7713", "7763", "7413"];
+
+fn current_gen() -> Result<SevGeneration> {
+    let mut bytestr = Vec::with_capacity(48);
+    let cpu_name = {
+        for cpuid in 0x8000_0002_u32..=0x8000_0004_u32 {
+            let cpuid = unsafe { x86_64::__cpuid(cpuid) };
+            let mut bytes: Vec<u8> = [cpuid.eax, cpuid.ebx, cpuid.ecx, cpuid.edx]
+                .iter()
+                .flat_map(|r| r.to_le_bytes().to_vec())
+                .collect();
+            bytestr.append(&mut bytes);
+        }
+
+        String::from_utf8(bytestr)
```

Comment:
Nitpicking: do you think it'd be possible to obtain the model number as an integer instead of a string?  If so, you could find the model by comparing integers instead of strings.

It's not super-important, because `sevctl ok` isn't a performance critical command, but it may be nice saving some CPU cycles there.

Reply:
Sure, that can be done.

While on the discussion, is there one central resource I can go to (provided by AMD I would assume) that would list all processors ID strings and which generation they support?

I'd like to ensure I don't miss any CPUs that are provided from AMD. I wish it was a bit easier to tell which generation a CPU was from a CPUID, but they don't explicitly tell which generation a user is currently running on.